src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateResolutionService.scala

```diff
@@ -86,7 +87,7 @@ class JobTemplateResolutionServiceImpl extends JobTemplateResolutionService {
 
   private def mergeSparkParameters(definitionParams: SparkDefinitionParameters, templateParams: SparkTemplateParameters): SparkInstanceParameters = {
     SparkInstanceParameters(
-      jobType = definitionParams.jobType,
+      jobType = JobTypes.Spark,
```

Comment:
Can't you put `jobType = JobTypes.Spark` into the `SparkInstanceParameters` case class as a default parameter, like for `ShellInstanceParameters`?